src/collections/componentSet.ts

```diff
@@ -292,6 +295,20 @@ export class ComponentSet extends LazyCollection<MetadataComponent> {
 
     const mdapiDeploy = new MetadataApiDeploy(operationOptions);
     await mdapiDeploy.start();
+
+    // Creates an array of unique metadata types to be deployed, uses Set to avoid duplicates.
+    const listOfMetadataTypesDeployed = Array.from(new Set(toDeploy.map((c) => c.type.name)));
+
+    void Lifecycle.getInstance().emitTelemetry({
+      eventName: 'metadata_api_deploy',
+      library: 'SDR',
+      function: 'ComponentSet.deploy',
+      totalNumberOfComponents: this.size,
```

Comment:
How is `totalNumberOfComponents` different in a significant way from `numberOfComponentsDeployed` at this point in the function?

Reply:
@shetzel I'm not entirely sure. The [original telemetry](https://github.com/salesforcecli/toolbelt/pull/215/files#diff-ca446f5ccf231e22c85434264adcb8d977f76d6375c1ae789fd39c1bde5aa324L313) used packages but since the concept of packages doesn't really exist in SDR I changed it to components.

I used `toDeploy` for `numberOfComponentsDeployed` because the variable name implies that's what being deployed and I couldn't figure out another way to identify that. If that's not the case though I can remove it and just leave `totalNumberOfComponents`.

If that's not the case and both of these numbers are the same I can just remove one.

Reply:
That original telemetry (iirc) was primarily added to gain some insight about how many customers/deploys were done as MPD (multiple package directory), and how big the projects were (number of types).  That 8000 number looks like it was originally for individual metadata but the code was changed to be about metadata types, otherwise it doesn't make sense.

I think we should change this a bit to be more useful.  Let's chat about it.